uwblueprint/social-diversity-for-children
# feat(i18n): migrate to v8 and get i18n to work

### Notion ticket link



[Setup Language Modal with i18n](https://www.notion.so/uwblueprintexecs/Setup-Language-Modal-with-i18n-f6718716cca74de5935b538dad1e7fe8)

https://user-images.githubusercontent.com/44826218/134756585-cf81780c-ea29-4f11-a7d7-3ad00dc4bf79.mp4



### Implementation description

- Migrate from i18n v7 to i18n v8 for better interfacing
- Hooked up language modal with i18n, samples on the index page with parent/volunteer
- Samples are in the index page. The language modal is no longer static
- We should now be ready for translations the second they come in :)



### Steps to test

1. Go to the home page of the website
2. Switch a language
3. The language switch label on the top right should switch to the correct language code (EN, ZH, JA, KO)
3. The parent/volunteer label should be translated (google translate ATM) to the appropriate language
4. Repeat steps 2 to 3 with all languages



### What should reviewers focus on?

- Design: Does the modal label make sense (simply using the capitalized language code)?

- Dev: Is i18n used appropriately (eg: does the config make sense)?
- Dev: Is the language modal safe and robust?

### Checklist
